sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2213,8 +2213,8 @@ case class Levenshtein(
   }
 
   override def inputTypes: Seq[AbstractDataType] = threshold match {
-    case Some(_) => Seq(StringType, StringType, IntegerType)
-    case _ => Seq(StringType, StringType)
+    case Some(_) => Seq(StringTypeAnyCollation, StringTypeAnyCollation, IntegerType)
+    case _ => Seq(StringTypeAnyCollation, StringTypeAnyCollation)
```

Comment:
Maybe we should special case this in CollationTypeCasts. We do not want to have an expression on the place of third parameter with StringType and take it into account, as it will explicitly be casted to Integer, and technically it's collation should not be counted. Something similar to If expression that is special cased.

Reply:
Added special case for Levenshtein to `CollationTypeCasts`, please check.